modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ExecutionMediator.java

```diff
@@ -25,9 +31,9 @@
    * Coordinates the execution of the {@code executor} using the given {@code context}
    *
    * @param executor an {@link OperationExecutor}
-   * @param context an {@link ExecutionContextAdapter}
-   * @return the operation's result
-   * @throws Exception if any exception is encountered
+   * @param context  an {@link ExecutionContextAdapter}
+   * @return a {@link Mono} with the operation's result
+   * @throws MuleException if the {@link Mono} could not be created
    */
-  Object execute(OperationExecutor executor, ExecutionContextAdapter context) throws Throwable;
+  Publisher<Object> execute(OperationExecutor executor, ExecutionContextAdapter context) throws MuleException;
```

Comment:
Is this the same method I said shouldn't throw an exception before, or a different one?